web-services/client/src/main/java/datawave/webservice/query/QueryImpl.java

```diff
@@ -413,6 +414,12 @@ public QueryImpl duplicate(String newQueryName) {
         if (null != this.parameters && !this.parameters.isEmpty())
             query.setParameters(new HashSet<Parameter>(this.parameters));
         query.setDnList(this.dnList);
+        if (null != this.optionalQueryParameters && !this.optionalQueryParameters.isEmpty()) {
```

Comment:
These were oversights in the duplicate method for a QueryImpl

Reply:
We use `StringUtils` for a null-safe check in many places, why not use `MapUtils` too?
```suggestion
        if (!MapUtils.isEmpty(this.optionalQueryParameters)) {
```